geode-core/src/main/java/org/apache/geode/internal/cache/tx/RemoteOperationMessage.java

```diff
@@ -169,14 +170,24 @@ public boolean checkDSClosing(ClusterDistributionManager dm) {
    */
   @Override
   public void process(final ClusterDistributionManager dm) {
+    InternalCache cache = getCache(dm);
+    if (cache == null) {
+      String message = getCacheClosedMessage(dm);
+      ReplyException replyException = new ReplyException(new CacheClosedException(message));
+      sendReply(getSender(), this.processorId, dm, replyException, null, 0);
+      return;
+    }
+    dm.getExecutors().getWaitingThreadPool().execute(() -> doRemoteOperation(dm, cache));
+  }
+
+  void doRemoteOperation(ClusterDistributionManager dm, InternalCache cache) {
     Throwable thr = null;
     boolean sendReply = true;
     LocalRegion r = null;
     long startTime = 0;
     try {
-      InternalCache cache = getCache(dm);
       if (checkCacheClosing(cache) || checkDSClosing(dm)) {
-        String message = "Remote cache is closed: " + dm.getId();
+        String message = getCacheClosedMessage(dm);
         if (cache == null) {
```

Comment:
I think it should no longer be possible for `cache` to be null at this point, given that we check if it's null in the `process()` method now.

Reply:
cache is not null, but it could be closed at this point.

Reply:
Agreed, so it's good to keep the `thr = cache.getCacheClosedException(message);` code, but it should be fine to remove the `thr = new CacheClosedException(message);` part and simplify this block a little.